### PR TITLE
Extract file data attributes

### PR DESCRIPTION
**Pull request by PrajwalM2212**
Signed-off-by: PrajwalM2212 <prajwalmmath@gmail.com>

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
I will modify the commit message after the PR has been completed.

**Comment by PrajwalM2212** [comment]
@nishakm @rnjudge  I have managed to parse `file_path, file_name, extattrs, checksum, checksum_type`

Example
```
./usr/local/go/pkg/linux_amd64/crypto
elliptic.a
-rw-r--r--|1000|1000|320462|1
83ee28cb7709861dad9501d0d21ccd544d986041471d43ef88ebd3a539235c2d
sha256
```

I can finish the PR after we have the function to add files in image layer.

**Comment by PrajwalM2212** [comment]
#### Part of output generated by running `layer.to_dict()`

'files': [{`'name': 'file2.txt', 'path': './documents/test/file2.txt'`, 'date': '', 'file_type': '', `'checksum_type': 'sha256', 'checksum': '9710f003d924890c7677b4dd91fd753f6ed71cc57d4f9482261b6786d81957fa',` 'version_control': '', 'version': '', `'extattrs': '-rw-r--r--|1000|1000|19|1'`, 'licenses': [], 'license_expressions': [], 'copyrights': [], 'authors': [], 'packages': [], 'urls': [], 'origins': []}, {`'name': 'file2.txt', 'path': './documents/test/test2/file2.txt'`, 'date': '', 'file_type': '', `'checksum_type': 'sha256', 'checksum': '885000512dee8ac814641bbf6a7c887012ec23a2fb3e3b2cff583c45a611317d'`, 'version_control': '', 'version': '', `'extattrs': '-rw-r--r--|1000|1000|29|1'`, 'licenses': [], 'license_expressions': [], 'copyrights': [], 'authors': [], 'packages': [], 'urls': [], 'origins': []}, {'name': `'file1.txt', 'path': './documents/test/test2/file1.txt'`, 'date': '', 'file_type': '', `'checksum_type': 'sha256', 'checksum': '885000512dee8ac814641bbf6a7c887012ec23a2fb3e3b2cff583c45a611317d'`, 'version_control': '', 'version': '',` 'extattrs': '-rw-r--r--|1000|1000|29|1'`, 'licenses': [], 'license_expressions': [], 'copyrights': [], 'authors': [], 'packages': [], 'urls': [], 'origins': []}, {`'name': 'file1.txt', 'path': './documents/test/file1.txt'`, 'date': '', 'file_type': '',` 'checksum_type': 'sha256', 'checksum': 'a3cccbc52486d50a86ff0bc1e6ea0e0b701ac4bb139f8713fa136ef9ec68e97e',` 'version_control': '', 'version': '', `'extattrs': '-rw-r--r--|1000|1000|20|1'`, 'licenses': [], 'license_expressions': [], 'copyrights': [], 'authors': [], 'packages': [], 'urls': [], 'origins': []}], 'origins': [], 'import_image': None, 'import_str': '', 'pkg_format': '', 'os_guess': '', 'files_analyzed': False, 'analyzed_output': ''}


@nishakm @rnjudge  This the output on the test image used in test_docker_image.  It correctly identifies all the four files present in the image. If this is okay, I will add a test in the test_docker_image, and submit the final PR.

**Comment by nishakm** [comment]
@PrajwalM2212 

> 'files': [{`'name': 'file2.txt', 'path': './documents/test/file2.txt'`, 'date': '', 'file_type': '', `'checksum_type': 'sha256', 'checksum': '9710f003d924890c7677b4dd91fd753f6ed71cc57d4f9482261b6786d81957fa',` 'version_control': '', 'version': '', `'extattrs': '-rw-r--r--|1000|1000|19|1'`, 'licenses': [], 'license_expressions': [], 'copyrights': [], 'authors': [], 'packages': [], 'urls': [], 'origins': []}, 

Only issue I see here is: `'path': './documents/test/file2.txt'` should be `'path': 'documents/test/file2.txt'`. Otherwise this looks good.

File: tern/classes/image_layer.py [hunk 1]
```diff
@@ -245,3 +246,24 @@ def gen_fs_hash(self):
...
+
+    def add_files(self):
+        fs_path = rootfs.get_untar_dir(self.__tar_file)
+        hash_file = os.path.join(os.path.dirname(fs_path), self.__fs_hash) + '.txt'
+        pattern = re.compile(r'([\w\-|]+)\s+([\w\.\/]+)')
+        with open(hash_file) as f:
+            content = f.readlines()
+        for line in content:
+            m = pattern.search(line)
+            if m:
```
Comment (nishakm) [inline tern/classes/image_layer.py]: The linter complains that there are too many local variables. I think you can use these values in the FileData assignment. Eg: `file_data = FileData(os.path.basename(m.group(2)), m.group(2))`

**Comment by nishakm** [comment]
@PrajwalM2212 turns out I need this change for #480 :) so I am marking it a high-priority. I hope you are OK if I take this change, modify it and submit it for you if you can't get to it by the end of the week.

**Comment by PrajwalM2212** [comment]
@nishakm I have made the changes you requested and also added the tests.

**Pull request merged**